src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/TaskActionsController.java

```diff
@@ -438,4 +454,19 @@ private boolean isUpdateCompletionProcessFlagEnabled(AccessControlResponse acces
             accessControlResponse.getUserInfo().getEmail()
         );
     }
+
+    /**
+     * Checks if the "WA_CANCELLATION_PROCESS_UPDATE" feature flag is enabled for the given user.
+     *
+     * @param accessControlResponse The access control response containing user information.
+     * @return true if the feature flag is enabled, false otherwise.
+     */
+    private boolean isUpdateCancellationProcessFlagEnabled(AccessControlResponse accessControlResponse) {
```

Comment:
I'd prefer to keep isUpdateCancellationProcessFlagEnabled and isUpdateCompletionProcessFlagEnabled to a dedicated class. That way it will be easy to test and decouple the logic.